js/views/invite_friends_dialog_view.js

```diff
@@ -14,11 +14,21 @@
 
       const convos = window.getConversations().models;
 
-      const friends = convos.filter(
-        d => !!d && d.isFriend() && d.isPrivate() && !d.isMe()
-      );
+      if (convo.isPublic()) {
+        this.friends = convos.filter(
+          d => !!d && d.isFriend() && d.isPrivate() && !d.isMe()
+        );
+      } else {
+        this.friends = convos.filter(
+          d =>
+            !!d &&
+            d.isFriend() &&
+            d.isPrivate() &&
+            !d.isMe() &&
+            !convo.get('members').includes(d.id)
```

Comment:
won't this break invites for opengroups?

Reply:
this is in the NOT open group case of the `if`

Reply:
ah my bad, sorry

Reply:
Couldn't you also do
```javascript
this.friends = convos.filter(
          d => !!d && d.isFriend() && d.isPrivate() && !d.isMe()
);
if (!convo.isPublic()) {
   const members = convo.get('members') || [];
    this.friends = this.friends.filter(d => !members.includes(d.id);
}
```

Less duplication